js/ui/scheduler/workspaces/view_data_provider.js

```diff
@@ -8,121 +8,97 @@ class ViewDataGenerator {
-        const isGroupedAllDayPanel = workspace.isGroupedAllDayPanel();
-
-        for(let groupIndex = 0; groupIndex < groupCount; ++groupIndex) {
-            let allDayPanelData = [];
-            let groupOffset = 0;
-            let renderRowCount = rowCount;
-            let viewCellsData = [];
-
-            if(groupIndex >= firstGroupIndex && groupIndex <= lastGroupIndex) {
-                const groupRowCount = (groupIndex + 1) * rowCountInGroup;
-                if(startRowIndex + renderRowCount > groupRowCount) {
-                    renderRowCount = groupRowCount - startRowIndex;
-                } else {
-                    const startGroupRowIndex = groupIndex * rowCountInGroup;
-                    if(startRowIndex < startGroupRowIndex && startRowIndex + renderRowCount > startGroupRowIndex) {
-                        groupOffset = startGroupRowIndex - startRowIndex;
-                        renderRowCount -= groupOffset;
-                    }
-                }
-
-                const needRenderAllDayPanel = ((startRowIndex + groupOffset) / rowCountInGroup) === groupIndex;
-                if(needRenderAllDayPanel) {
-                    allDayPanelData = this._generateAllDayPanelData(options, groupIndex, rowCount, cellCount);
-                }
 
-                viewCellsData = this._generateViewCellsData(
-                    options,
-                    renderRowCount,
-                    startRowIndex,
-                    groupOffset
-                );
-            }
+        const viewDataMap = [];
+        for(let groupIndex = 0; groupIndex < groupCount; groupIndex += 1) {
+            const allDayPanelData = this._generateAllDayPanelData(options, groupIndex, rowCount, cellCount);
+            const viewCellsData = this._generateViewCellsData(
+                options,
+                rowCount,
+                0,
+                rowCount * groupIndex
+            );
 
-            viewCellsData.length && groupedData.push({
-                groupIndex: groupIndex,
-                dateTable: viewCellsData,
-                allDayPanel: allDayPanelData,
-                isGroupedAllDayPanel
-            });
+            allDayPanelData && viewDataMap.push(allDayPanelData);
+            viewDataMap.push(...viewCellsData);
         }
 
-        return {
-            viewData: {
-                groupedData,
-                isVirtual: true,
-                topVirtualRowHeight,
-                bottomVirtualRowHeight,
-                cellCountInGroupRow,
-            },
-        };
+        return viewDataMap;
+    }
+
+    _generateViewDataMap(completeViewDataMap, options) {
+        const { startRowIndex, rowCount } = options;
+
+        const isVerticalGrouping = this.workspace._isVerticalGroupedWorkSpace();
+        const showAllDayPanel = this.workspace._isShowAllDayPanel();
+
+        const indexDifference = isVerticalGrouping || (!isVerticalGrouping && !showAllDayPanel) ? 0 : 1;
```

Comment:
Can we simplify to:
`const indexDifference = isVerticalGrouping || !showAllDayPanel ? 0 : 1;`